src/lib.rs

```diff
@@ -236,21 +288,59 @@ impl<B: MysqlShim<W>, R: Read, W: Write> MysqlIntermediary<B, R, W> {
     }
 
     fn init(&mut self) -> Result<(), B::Error> {
+        // https://dev.mysql.com/doc/internals/en/connection-phase-packets.html#packet-Protocol::HandshakeV10
         self.writer.write_all(&[10])?; // protocol 10
 
-        // 5.1.10 because that's what Ruby's ActiveRecord requires
-        self.writer.write_all(&b"5.1.10-alpha-msql-proxy\0"[..])?;
+        self.writer.write_all(self.shim.version().as_bytes())?;
+        self.writer.write_all(&[0x00])?;
+
+        // connection_id (4 bytes)
+        self.writer
+            .write_all(&self.shim.connect_id().to_le_bytes())?;
+
+        let server_capabilities = (
+            CapabilityFlags::CLIENT_PROTOCOL_41
+                | CapabilityFlags::CLIENT_SECURE_CONNECTION
+                | CapabilityFlags::CLIENT_PLUGIN_AUTH
+                | CapabilityFlags::CLIENT_PLUGIN_AUTH_LENENC_CLIENT_DATA
+                | CapabilityFlags::CLIENT_CONNECT_WITH_DB
+                | CapabilityFlags::CLIENT_RESERVED
+            // | CapabilityFlags::CLIENT_SSL
+        )
+            .bits();
 
-        self.writer.write_all(&[0x08, 0x00, 0x00, 0x00])?; // TODO: connection ID
-        self.writer.write_all(&b";X,po_k}\0"[..])?; // auth seed
-        self.writer.write_all(&[0x00, 0x42])?; // just 4.1 proto
+        let server_capabilities = server_capabilities.to_le_bytes();
+        let default_auth_plugin = self.shim.default_auth_plugin();
+        let scramble = self.shim.salt();
+
+        self.writer
+            .write_all(&scramble[0..AUTH_PLUGIN_DATA_PART_1_LENGTH])?; // auth-plugin-data-part-1
+        self.writer.write_all(&[0x00])?;
+
+        self.writer.write_all(&server_capabilities[..2])?; // The lower 2 bytes of the Capabilities Flags, 0x42
+                                                           // self.writer.write_all(&[0x00, 0x42])?;
         self.writer.write_all(&[0x21])?; // UTF8_GENERAL_CI
-        self.writer.write_all(&[0x00, 0x00])?; // status flags
-        self.writer.write_all(&[0x00, 0x00])?; // extended capabilities
-        self.writer.write_all(&[0x00])?; // no plugins
-        self.writer.write_all(&[0x00; 6][..])?; // filler
-        self.writer.write_all(&[0x00; 4][..])?; // filler
-        self.writer.write_all(&b">o6^Wz!/kM}N\0"[..])?; // 4.1+ servers must extend salt
+        self.writer.write_all(&[0x00, 0x00])?; // status_flags
+        self.writer.write_all(&server_capabilities[2..4])?; // The upper 2 bytes of the Capabilities Flags
+
+        if default_auth_plugin.is_empty() {
```

Comment:
This doesn't seem right. Should we make `default_auth_plugin` return an `Option` instead and check `None` here?